src/client/components/form-editor/entity-form.js

```diff
@@ -15,70 +15,110 @@ class EntityForm extends DirtyComponent {
   constructor(props) {
     super(props);
     this.state = this.data = _.assign( {
-      style: 'form-entity',
-      showEntityInfo: false,
+      style: 'form-entity'
     }, props );
 
 
     if(this.data.entity){
       this.data.entity.on("complete", () => {
+
         this.mergeWithOtherEntities();
         this.dirty();
       });
     }
 
     this.entityInfoClasses = ".entity-info-section, .entity-info-progression";
 
+
+    if(this.data.entity.completed())
+      this.iconType = 'arrow_drop_up';
+    else
+      this.iconType = 'arrow_drop_down';
   }
 
   componentDidMount(){
-    // this.hideEntityInfo();
     emitter.on('esc', () => this.hideEntityInfo());
-  }
 
-  toggleEntityInfo(){
-    let target = ReactDom.findDOMNode(this);
-    let entityInfoSections = target.querySelectorAll(this.entityInfoClasses);
-    entityInfoSections.forEach(ei => {
-      ei.style.visibility = ei.style.visibility === "hidden" ? "visible": "hidden" ;
-      if(ei.style.visibility === "hidden") {
-        ei.style.height = 0;
-      }
-      else {
-        ei.style.height = "100%";
+    this.state.bus.on('showEntityInfo', ()=>{
+      this.showEntityInfo();
+    });
 
-      }
+
+    this.state.bus.on('hideEntityInfo', ()=>{
+      this.hideEntityInfo();
     });
-    this.dirty();
-  }
 
-  hideEntityInfo(){
+    this.state.entity.on('associated', ()=>{
```

Comment:
Does this code mean that when the entity is associated, change the style of this component?

Reply:
If a clone of this entity's association is updated, its information needs to be updated as well. I could only get the EntityInfo component to refresh when its style is updated.

Reply:
This seems like it should just be using a flag for info visibility that should be checked in render()

Reply:
That's not about visibility though. It's about re-rendering a component inside the render function.

Reply:
Either:

(1) You need to pass props to the child component that changes that component itself.

(2) The subcomponent itself needs to know when to change internally.

This seems like a form-only issue, so it's probably (1).

More importantly, though, why would the form need to modify an EntityInfo?  Each EntityInfo is stand-alone.

Reply:
Ideally, you would expect (2), but it doesn't happen. EntityInfo takes Entity as its main prop, and whenever an entity's association is updated, EntityInfo is supposed to update its rendering. 
As I explained above, it is the clones that cause the problem. Say we have two interactions "A phosphorylates B" and "A activates expression of C", which are already grounded. Suppose the user decides to change the grounding of A, by interacting with the A in the phosphorylation interaction.  The second A's grounding information needs to be concurrently updated, which requires a re-rendering of its EntityInfo, which doesn't happen on its own.

Reply:
I mean, yes these suggestions are great in theory and they are the obvious explanations, but they don't help solve the problem. 
Of course we only observe this in the form because clone entities are shown as separate entity boxes on the form as opposed to the graph editor where the same node can take place in more than one interactions.

Reply:
I don't know if there is a value; maybe we can ask about this in the user tests. I would prefer to be able to see them all at once as a user, but then I'm not a biologist.

Showing a single EntityInfo may simplify things. Still, I think this is an issue that remains to be discussed, even if at a later iteration.